### ISSUE TITLE
Memory Leak in Transform

### ISSUE THREAD

**Issue by MariaMikhnevich** [konvajs/konva#1872]
In Transform, in the `tr.nodes` (`setNodes`) method, one of the subscriptions depends on the private property `_mattersAffectingSize`. `Node` has the value of this property equal to `[]`, and only some elements of the array have values.

![image](https://github.com/user-attachments/assets/2257e160-dd62-4966-b75c-e9500bf0ed1b)

![image](https://github.com/user-attachments/assets/9bf84b12-6335-4ee6-b7d5-2492253954e8)

Upon `tr.detach()`, subscriptions are unsubscribed only with the value. But due to the fact that our `_attrsAffectingSize` property has the value `[]`, the map does not start and the `additionalEvents` variable is assigned the value `"`. Accordingly, when using `tr.destroy()`, this subscription is not unsubscribed, and if we create another transformer and add the same element to nodes, and then call `tr.destroy()`, the number of subscriptions in the element increases.
![Screenshot from 2025-01-09 15-54-45](https://github.com/user-attachments/assets/31b7348a-09bf-4028-addb-020f1c68be66)

![image](https://github.com/user-attachments/assets/5d3fda2e-f0bb-4b92-a3d0-c81ba63cc6d6)

![image](https://github.com/user-attachments/assets/7c817c87-a464-40c2-ba29-1dc954186b2e)

![image](https://github.com/user-attachments/assets/40d05ca1-4ab4-43b3-b6a5-291507ab3494)

If you set the `_attrsAffectingSize` property in `Node` to `["]` instead of `[]`, subscriptions and unsubscriptions start working correctly.
![image](https://github.com/user-attachments/assets/f66b223a-fd30-4f29-9155-df410b631704)
![image](https://github.com/user-attachments/assets/55bae12d-0398-41f5-93e2-e86ecc062286)

**Comment by lavrton** [comment]
Do you want to create a Pull Request with the tests and fix?

**Comment by MariaMikhnevich** [comment]
Thanks for the answer! Sorry, I'm not good at tests. Could you do it yourself?

**Comment by MariaMikhnevich** [comment]
But I can do a Pull Request without tests. Do it?
